### PR TITLE
Add Blob column to website

### PR DESCRIPTION
**Pull request by avalonche**
## 📝 Summary

Adds a blob column to the website to show the number of blobs in the block. On hover will also indicate the amount of blob gas used. Also some refactoring of the block validation json marshalling. 

Deployed on goerli.

Screenshot 2024-01-30 at 4 13 07 pm

## ⛱ Motivation and Context

Allows easy monitoring of blobs in blocks. 

## 📚 References

See also https://github.com/flashbots/relay-specs/issues/31 for exposing the same data on the data API. These changes will allow the same information easily exposed to the data API.

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [x] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

**Comment by metachris** [comment]
V cool! Please rebase.

Wonder if we should slim down the column title from "Num blobs" to only "Blobs". I think yes.

File: common/types.go [hunk 1]
```diff
@@ -326,25 +326,93 @@ func (b *BidTraceV2WithTimestampJSON) ToCSVRecord() []string {
 	}
 }
 
+type BidTraceV2WithBlobFields struct {
+	builderApiV1.BidTrace
```
Comment (metachris) [inline common/types.go:330]: could extend V2

File: common/types.go [hunk 2]
```diff
@@ -326,25 +326,93 @@ func (b *BidTraceV2WithTimestampJSON) ToCSVRecord() []string {
...
+	GasUsed              uint64 `json:"gas_used,string"`
+	Value                string `json:"value"`
+	NumTx                uint64 `json:"num_tx,string"`
+	BlockNumber          uint64 `json:"block_number,string"`
+	NumBlobs             uint64 `json:"num_blobs,string"`
+	BlobGasUsed          uint64 `json:"blob_gas_used,string"`
+	ExcessBlobGas        uint64 `json:"excess_blob_gas,string"`
+}
+
+func (b BidTraceV2WithBlobFields) MarshalJSON() ([]byte, error) {
```
Comment (metachris) [inline common/types.go:355]: why does it need custom marshal and unmarshal functions?

File: common/types.go [hunk 3]
```diff
@@ -326,25 +326,93 @@ func (b *BidTraceV2WithTimestampJSON) ToCSVRecord() []string {
...
+	if err != nil {
+		return err
+	}
+	b.BidTrace = *bidTrace
+	return nil
+}
+
 type BlockSubmissionInfo struct {
 	BidTrace                   *builderApiV1.BidTrace
-	Slot                       uint64
```
Comment (metachris) [inline common/types.go:331]: why remove this and other info fields?

Comment (metachris) [reply common/types.go:331]: oh i see, because it's already inside BidTrace

File: common/types.go [hunk 4]
```diff
@@ -326,25 +326,93 @@ func (b *BidTraceV2WithTimestampJSON) ToCSVRecord() []string {
...
-	BlockHash                  phase0.Hash32
-	ParentHash                 phase0.Hash32
 	ExecutionPayloadBlockHash  phase0.Hash32
 	ExecutionPayloadParentHash phase0.Hash32
-	Builder                    phase0.BLSPubKey
-	Proposer                   phase0.BLSPubKey
-	ProposerFeeRecipient       bellatrix.ExecutionAddress
 	GasUsed                    uint64
```
Comment (metachris) [inline common/types.go:405]: GasUsed and GasLimnit is also part of BidTrace

**Review by metachris — APPROVED** [review]
nice work! left a few questions, overall lgtm!

File: common/types.go (hunk 4 above)
Comment (avalonche) [reply common/types.go:405]: The fields are using the gas used and gas limit from the execution payload

File: common/types.go (hunk 1 above)
Comment (avalonche) [reply common/types.go:330]: This would expose the blob data to the data api as it uses the same json structs

File: common/types.go (hunk 2 above)
Comment (avalonche) [reply common/types.go:355]: The bid trace is an embedded struct and json.Marshal does not marshal properly for embedded structs.

**Pull request merged**